records/validation_test.go

```diff
@@ -58,15 +44,22 @@ func TestValidateResolvers(t *testing.T) {
 		{[]string{"2001:0db8:3c4d:0015:0000:0000:1a2f:1a2b"}, true},
 		{[]string{"2001:db8:3c4d:15::1a2f:1a2b"}, true},
 		{[]string{"2001:0db8:3c4d:0015:0000:0000:1a2f:1a2b", "2001:db8:3c4d:15::1a2f:1a2b"}, false},
+	} {
+		validate(t, i+1, tc, validateResolvers)
 	}
-	for i, tc := range table {
-		err := validateResolvers(tc.rs)
-		if (err == nil && tc.valid) || (err != nil && !tc.valid) {
-			continue
-		} else if tc.valid {
-			t.Fatalf("test %d failed, unexpected error validating resolvers %v: %v", i+1, tc.rs, err)
-		} else {
-			t.Fatalf("test %d failed, expected validation error for resolvers(%d) %v", i+1, len(tc.rs), tc.rs)
-		}
+}
+
+type validationTest struct {
```

Comment:
I tend to put those utilities on-top. But maybe I am just used to those ancient declare-before-use languages.